inc/Prognoser.h

```diff
@@ -31,108 +26,42 @@
-         *   Directs the main prognostics loop- runs until monitor->stop()
+         *  configuration file
          */
-        void run() override;
-
-        /// Save the current state to the prognostic history file
-        void saveState() const;
-
-        /// Load the last state from the prognostic history file
-        void loadHistory();
-
-        /// Reset the prognostic history file (incase of maintanance, etc.)
-        void resetHistory() const;
+		Prognoser(ConfigMap& ) {};
+		
+		Prognoser() = default;
 
         //*-------------------------------------------------------*
         //|     Virtual methods - Used by Component Prognoser     |
         //*-------------------------------------------------------*
 
-        /** @brief     check the validity of any input (sensor) data.
-         *
-         *             This could be as simple as bound checks or a complicated
-         *             analysis. By default this is not done- making this step
-         *             optional in the component prognoser implementation
-         */
-        virtual void checkInputValidity() {}
-
-        /** @brief     check if there is enough new data to preform prognosis
-         *  @return    if there is enough data
-         *
-         *             Check if the data exists and is new enough to be used for
-         *             prognosis. If false is returned prognostics steps will be
-         *             skipped. By default this returns true- making this step
-         *             optional in the component prognoser implementation
-         */
-        virtual bool isEnoughData() const {
-            return true;
-        }
-
         /** @brief     Prognostic Step
          *
          *             Preform model updates. This is done every step where there is
          *             enough data. This is a required method in any component
          *             prognoser
          */
-        virtual void step() = 0;
-
-        /** @brief     check the validity of any prognostics results.
-         *
-         *             This could be as simple as bound checks or a complicated
-         *             analysis. By default this a simple bounds test on timeToEvent
-         *             - making this step optional in the component prognoser
-         *             implementation
-         *             Default implemented in Prognoser
-         */
-        virtual void checkResultValidity();
-
-    protected:
-        Datum<double> getValue(MessageId key);
-
-    private:
-        std::string histFileName; ///< Name of history file
-        std::vector<std::string> histStr; ///< Current contents of history file
-
-        unsigned int loopInterval; ///< Time between prognostic loops (ms)
-        unsigned int saveInterval; ///< Loops between saves
-        bool saveEnabled;
-
-        std::map<MessageId, std::function<Datum<double>(void)>> lookup;
+        virtual Prediction step(std::map<MessageId, Datum<double> > data) = 0;
```

Comment:
Remove space between closing angle brackets

Reply:
I refuse to believe that's actually true. I submit we should remove the space anyway specifically because I want to see this error if it actually exists. 🤣